nntrainer/engine.h

```diff
@@ -0,0 +1,255 @@
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLayerObject(int_key);
+  }
+
+  /**
+   * @brief Create an Optimizer Object with Optimizer name
+   *
+   * @param type Optimizer name
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Optimizer object
+   */
+  std::unique_ptr<nntrainer::Optimizer>
+  createOptimizerObject(const std::string &type,
+                        const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createOptimizerObject(type);
+  }
+
+  /**
+   * @brief Create an Optimizer Object with Optimizer key
+   *
+   * @param int_key key
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Optimizer object
+   */
+  std::unique_ptr<nntrainer::Optimizer>
+  createOptimizerObject(const int int_key,
+                        const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createOptimizerObject(int_key);
+  }
+
+  /**
+   * @brief Create an LearningRateScheduler Object with type
+   *
+   * @param type type of LearningRateScheduler
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the LearningRateScheduler object
+   */
+  std::unique_ptr<ml::train::LearningRateScheduler>
+  createLearningRateSchedulerObject(
+    const std::string &type,
+    const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLearningRateSchedulerObject(type, properties);
+  }
+
+  /**
+   * @brief Create an LearningRateScheduler Object with key
+   *
+   * @param int_key key
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the LearningRateScheduler object
+   */
+  std::unique_ptr<ml::train::LearningRateScheduler>
+  createLearningRateSchedulerObject(
+    const int int_key, const std::vector<std::string> &properties = {}) {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLearningRateSchedulerObject(int_key, properties);
+  }
+
+  /**
+   * @brief Get Working Path from a relative or representation of a path
+   * starting from @a working_path_base.
+   * @param[in] path to make full path
+   * @return If absolute path is given, returns @a path
+   * If relative path is given and working_path_base is not set, return
+   * relative path.
+   * If relative path is given and working_path_base has set, return absolute
+   * path from current working directory
+   */
+  const std::string getWorkingPath(const std::string &path = "");
+
+  /**
+   * @brief Set Working Directory for a relative path. working directory is set
+   * canonically
+   * @param[in] base base directory
+   * @throw std::invalid_argument if path is not valid for current system
+   */
```

Comment:
I don't really understand what does it mean to set WD for a relative path.
Also I do know what the cannonical path is (as defined by Posix), but not in context of second sentence in `@brief`, I didn't get what does it mean to set working directory cannonicaly